java/src/main/kotlin/io/spine/validation/java/generate/option/RangeFieldGenerator.kt

```diff
@@ -0,0 +1,224 @@
+    private val fieldType = field.type
+    private val declaringType = field.declaringType
+    private val getter = message.field(field).getter<Any>()
+    private val lower = view.lowerBound
+    private val upper = view.upperBound
+
+    /**
+     * Generates code for a field represented by the [view].
+     */
+    @Suppress("UNCHECKED_CAST") // The cast is guaranteed due to the field type checks.
+    override fun generate(): FieldOptionCode = when {
+        fieldType.isSingular -> checkWithinTheRange(getter as Expression<Number>)
+
+        fieldType.isList ->
+            CodeBlock(
+                """
+                for (var element : $getter) {
+                    ${checkWithinTheRange(ReadVar("element"))}
+                }
+                """.trimIndent()
+            )
+
+        else -> error(
+            "The field type `${fieldType.name}` is not supported by `RangeFieldGenerator`." +
+                    " Please ensure that the supported field types in this generator match those" +
+                    " used by `RangePolicy` when validating the `RangeFieldDiscovered` event."
+        )
+    }.run { FieldOptionCode(this) }
+
+    /**
+     * Returns a [CodeBlock] that checks that the given [value] is within the [lower]
+     * and [upper] bounds.
+     *
+     * If the passed value is out of the range, creates an instance of [ConstraintViolation]
+     * adding it to the [violations] list.
+     */
+    private fun checkWithinTheRange(value: Expression<Number>): CodeBlock =
+        CodeBlock(
+            """
+            if (${doesNotBelongToRange(value)}) {
+                var fieldPath = ${parentPath.resolve(field.name)};
+                var typeName =  ${parentName.orElse(declaringType)};
+                var violation = ${violation(ReadVar("fieldPath"), ReadVar("typeName"), value)};
+                $violations.add(violation);
+            }    
+            """.trimIndent()
+        )
+
+    /**
+     * Returns a boolean expression that checks if the given [value] is within
+     * the [lower] and [upper] bounds.
+     *
+     * Unsigned values are handled in a special way because Java does not support them natively.
+     * [IntegerClassName] and [LongClassName] classes provide static methods to treat signed
+     * types as unsigned, including parsing, printing, comparison and math operations.
+     * Outside of these methods, these primitives remain just signed types.
+     */
+    private fun doesNotBelongToRange(value: Expression<Number>): Expression<Boolean>  {
+        val valueCase = lower.valueCase // This case is the same for both `lower` and `upper`.
+        val lowerLiteral = lower.asLiteral()
+        val lowerOperator = if (lower.inclusive) "<" else "<="
+        val upperLiteral = upper.asLiteral()
+        val upperOperator = if (upper.inclusive) ">" else ">="
+        if (valueCase in listOf(UINT32_VALUE, UINT64_VALUE)) {
+            Compilation.warning(view.file, field.span) {
```

Comment:
@yevhenii-nadtochii, 

We can probably store a number of times we issue this warning and put the full text only once. On second and after occasions we just print a short text referring to the same issue having some text sounding like "Already reported".

Gradle does a similar thing for repeated errors.

This way the console log won't be "polluted".